esptool.py

```diff
@@ -2148,6 +2148,17 @@ def __call__(self, parser, namespace, values, option_string=None):
             except IndexError:
                 raise argparse.ArgumentError(self,'Must be pairs of an address and the binary filename to write there')
             pairs.append((address, argfile))
+
+        # Sort the addresses and check for overlapping
+        end = 0
+        for address, argfile in sorted(pairs):
+            argfile.seek(0,2)  # seek to end
+            size = argfile.tell()
```

Comment:
Sorry, my bad. The calculation should be better now and there are two small unit tests related to overlapping.